src/actions/arcActions.ts

```diff
@@ -612,30 +609,44 @@ export function voteOnProposal(daoAvatarAddress: string, proposalId: string, vot
 
       const voteTransaction = await votingMachineInstance.vote({ proposalId, vote });
     } catch (err) {
-      dispatch(showAlert(('Voting failed: ' + err.message)));
       dispatch({ type: arcConstants.ARC_VOTE_REJECTED, payload: err.message });
     }
   };
 }
 
+export type StakeAction = IAsyncAction<'ARC_STAKE', {
+  avatarAddress: string,
+  proposalId: string,
+  stake: number,
+  prediction: VoteOptions,
+  stakerAddress: string,
+}, {
+  proposal: any,
+  stake: any,
+}>
+
 export function stakeProposal(daoAvatarAddress: string, proposalId: string, prediction: number, stake: number) {
   return async (dispatch: Redux.Dispatch<any>, getState: () => IRootState) => {
     const web3: Web3 = Arc.Utils.getWeb3();
     const currentAccountAddress: string = getState().web3.ethAccountAddress;
-
-    // TODO: num transactions pending...
-    let payload: any = {
-      stake: {
-        avatarAddress: daoAvatarAddress,
-        proposalId,
-        stake,
-        prediction,
-        stakerAddress: currentAccountAddress,
-        transactionState: TransactionStates.Unconfirmed,
-      },
+    const proposal: IProposalState = getState().arc.proposals[proposalId];
+
+    const meta = {
+      avatarAddress: daoAvatarAddress,
+      proposalId,
+      stake,
+      prediction,
+      stakerAddress: currentAccountAddress,
     };
 
-    dispatch({ type: arcConstants.ARC_STAKE_PENDING, payload });
+    dispatch({
+      type: arcConstants.ARC_STAKE,
+      sequence: AsyncActionSequence.Pending,
+      operation: {
```

Comment:
aren't there two steps for staking? so shouldn't we say that here?

Reply:
Currently both transactions (approve and transfer) happen inside `arc.js` in the `stake` function. So I can't really put something after the approve and before the transfer.
Probably needs some solution in arc.js.

Reply:
but we want to pass in the totalSteps at the beginning (2) so we know how many more there are to come

Reply:
If I put `totalSteps: 2` and have only one `pending` action it will seem weird that it succeeds while showing that it only did 1 step out of 2. So until we can dispatch an action after the approve and before the transfer, I think it's better leaving it this way.

Reply:
added `totalSteps: 1`

Reply:
well might be more confusing to have steps = 1 when there really are two steps, so maybe leave out for now?

Reply:
removed in https://github.com/daostack/alchemy/pull/46/commits/efae6596a110cca203d88f089cb58452f384127f